pkgs/development/compilers/rust/rustc.nix

```diff
@@ -59,28 +52,69 @@ stdenv.mkDerivation {
 
   # We need rust to build rust. If we don't provide it, configure will try to download it.
   # Reference: https://github.com/rust-lang/rust/blob/master/src/bootstrap/configure.py
-  configureFlags = configureFlags
-                ++ [ "--enable-local-rust" "--local-rust-root=${rustPlatform.rust.rustc}" "--enable-rpath"
-                     "--enable-vendor"
-                     "--jemalloc-root=${prefixedJemalloc}/lib"
-                     "--default-linker=${targetPackages.stdenv.cc}/bin/cc" ]
-                ++ optional (!withBundledLLVM) [ "--enable-llvm-link-shared" "--llvm-root=${llvmShared}" ]
-                ++ optional (targets != []) "--target=${target}";
+  configureFlags = let
+    setBuild = "--set=target.${stdenv.hostPlatform.config}";
+    ccForBuild = "${buildPackages.stdenv.cc}/bin/{buildPackages.stdenv.cc.targetPrefix}cc";
+    setHost = "--set=target.${stdenv.hostPlatform.config}";
+    ccForHost = "${stdenv.cc}/bin/${stdenv.cc.targetPrefix}cc";
+    setTarget = "--set=target.${stdenv.targetPlatform.config}";
+    ccForTarget = "${targetPackages.stdenv.cc}/bin/${targetPackages.stdenv.cc.targetPrefix}cc";
+  in [
+    "--release-channel=stable"
+    "--enable-local-rust"
+    "--local-rust-root=${rustPlatform.rust.rustc}"
+    "--enable-rpath"
+    "--enable-vendor"
+    "--default-linker=${ccForBuild}"
+    "--build=${stdenv.buildPlatform.config}"
+    "--host=${stdenv.hostPlatform.config}"
+    "--target=${stdenv.targetPlatform.config}"
+    "${setBuild}.cc=${ccForBuild}"
+    "${setHost}.cc=${ccForHost}"
+    "${setTarget}.cc=${ccForTarget}"
+
+    "${setBuild}.linker=${ccForBuild}"
+    "${setHost}.linker=${ccForHost}"
+    "${setTarget}.linker=${ccForTarget}"
+
+    "${setBuild}.cxx=${buildPackages.stdenv.cc}/bin/c++"
+    "${setHost}.cxx=${buildPackages.stdenv.cc}/bin/c++"
+    "${setTarget}.cxx=${targetPackages.stdenv.cc}/bin/${targetPackages.stdenv.cc.targetPrefix}c++"
+
+    "${setBuild}.jemalloc=${hostJemalloc}/lib/libjemalloc_pic.a"
+    "${setHost}.jemalloc=${hostJemalloc}/lib/libjemalloc_pic.a"
+    "${setTarget}.jemalloc=${targetJemalloc}/lib/libjemalloc_pic.a"
```

Comment:
@Ericson2314 this one seems actually undefined, when I try to build `nix-build default.nix -A pkgsCross.aarch64-multiplatform.rustc`

Reply:
Yeah that's OK. For now. I need to change something in booter.nix to avoid these issues. Need to throw in a `buildPackages` to get the cross compiler vs cross compiler compiler, which is the immediate need anyways.